application-templates/starter/src/components/channel-details/channel-details.js

```diff
@@ -0,0 +1,139 @@
+  });
+  const showNotification = useShowNotification();
+  const showApiErrorNotification = useShowApiErrorNotification();
+  const channelDetailsUpdater = useChannelDetailsUpdater();
+  const handleSubmit = useCallback(
+    async (formikValues, formikHelpers) => {
+      const data = formValuesToDoc(formikValues);
+      try {
+        await channelDetailsUpdater.execute({
+          originalDraft: channel,
+          nextDraft: data,
+        });
+        showNotification({
+          kind: 'success',
+          domain: DOMAINS.SIDE,
+          text: intl.formatMessage(messages.channelUpdated, {
+            channelName: formatLocalizedString(formikValues, {
+              key: 'name',
+              locale: dataLocale,
+              fallbackOrder: projectLanguages,
+            }),
+          }),
+        });
+      } catch (graphQLErrors) {
+        const transformedErrors = transformErrors(graphQLErrors);
+        if (transformedErrors.unmappedErrors.length > 0)
+          showApiErrorNotification({
+            errors: transformedErrors.unmappedErrors,
+          });
+
+        formikHelpers.setErrors(transformedErrors.formErrors);
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [channelDetailsUpdater]
+  );
+
+  return (
+    <ChannelsDetailsForm
+      initialValues={docToFormValues(channel, projectLanguages)}
+      onSubmit={handleSubmit}
+    >
+      {(formProps) => {
+        return (
+          <FormModalPage
+            title={formatLocalizedString(
+              {
+                name: transformLocalizedFieldToLocalizedString(
+                  channel?.nameAllLocales ?? []
+                ),
+              },
+              {
+                key: 'name',
+                locale: dataLocale,
+                fallbackOrder: projectLanguages,
+                fallback: NO_VALUE_FALLBACK,
+              }
+            )}
+            isOpen
+            onClose={props.onClose}
+            isPrimaryButtonDisabled={
+              formProps.isSubmitting || !formProps.isDirty || !canManage
+            }
+            isSecondaryButtonDisabled={!formProps.isDirty}
+            onSecondaryButtonClick={formProps.resetForm}
+            onPrimaryButtonClick={formProps.submitForm}
+            labelPrimaryButton={FormModalPage.Intl.save}
+            labelSecondaryButton={FormModalPage.Intl.revert}
+          >
+            {loading && (
+              <Spacings.Stack alignItems="center">
+                <LoadingSpinner />
+              </Spacings.Stack>
+            )}
+            {error ? (
+              <ContentNotification type="error">
+                <Text.Body>
+                  {intl.formatMessage(messages.channelDetailsErrorMessage)}
+                </Text.Body>
+              </ContentNotification>
+            ) : channel === null ? (
+              <PageNotFound />
+            ) : null}
+            {channel && formProps.formElements}
```

Comment:
```suggestion
            {error && (
              <ContentNotification type="error">
                <Text.Body>
                  {intl.formatMessage(messages.channelDetailsErrorMessage)}
                </Text.Body>
              </ContentNotification>
            )}
            {channel ? formProps.formElements : (
              <PageNotFound />
            )}
```

Reply:
Thank you 🙏 I edited it a little due to `<PageNotFound />` flickering on loading